stubs/paho-mqtt/paho/mqtt/client.pyi

```diff
@@ -0,0 +1,321 @@
+MQTT_ERR_INVAL: int
+MQTT_ERR_NO_CONN: int
+MQTT_ERR_CONN_REFUSED: int
+MQTT_ERR_NOT_FOUND: int
+MQTT_ERR_CONN_LOST: int
+MQTT_ERR_TLS: int
+MQTT_ERR_PAYLOAD_SIZE: int
+MQTT_ERR_NOT_SUPPORTED: int
+MQTT_ERR_AUTH: int
+MQTT_ERR_ACL_DENIED: int
+MQTT_ERR_UNKNOWN: int
+MQTT_ERR_ERRNO: int
+MQTT_ERR_QUEUE_SIZE: int
+MQTT_ERR_KEEPALIVE: int
+MQTT_CLIENT: int
+MQTT_BRIDGE: int
+MQTT_CLEAN_START_FIRST_ONLY: int
+sockpair_data: bytes
+_Payload: TypeAlias = str | bytes | bytearray | float
+_ExtraHeader: TypeAlias = dict[str, str] | Callable[[dict[str, str]], dict[str, str]]
+_OnLog: TypeAlias = Callable[[Client, Any, int, str], None]
+_OnConnect = TypeVar(
+    "_OnConnect",
+    Callable[[Client, Any, dict[str, int], int], None],
+    Callable[[Client, Any, dict[str, int], ReasonCodes, Properties | None], None],
+)
+_OnConnectFail: TypeAlias = Callable[[Client, Any], None]
+_OnSubscribe = TypeVar(
+    "_OnSubscribe",
+    Callable[[Client, Any, int, Iterable[int]], None],
+    Callable[[Client, Any, Iterable[ReasonCodes], Iterable[Properties]], None],
+)
+_OnMessage: TypeAlias = Callable[[Client, Any, MQTTMessage], None]
+_OnPublish: TypeAlias = Callable[[Client, Any, int], None]
+_OnUnsubscribe = TypeVar(
+    "_OnUnsubscribe",
+    Callable[[Client, Any, int], None],
+    Callable[[Client, Any, int, Iterable[Properties], Iterable[ReasonCodes]], None],
+)
+_OnDisconnect = TypeVar(
+    "_OnDisconnect", Callable[[Client, Any, int], None], Callable[[Client, Any, int, ReasonCodes | None, Properties | None], None]
+)
+_OnSocket: TypeAlias = Callable[[Client, Any, _socket.socket | Any | None], None]
+
+class WebsocketConnectionError(ValueError): ...
+
+def error_string(mqtt_errno: int) -> str: ...
+def connack_string(connack_code: int) -> str: ...
+def base62(num: int, base: str = ..., padding: int = ...) -> str: ...
+def topic_matches_sub(sub: str, topic: str) -> bool: ...
+
+class MQTTMessageInfo:
+    mid: int
+    rc: int
+    def __init__(self, mid: int) -> None: ...
+    def __iter__(self) -> MQTTMessageInfo: ...
+    def __next__(self) -> int: ...
+    def next(self) -> int: ...
+    def __getitem__(self, index: int) -> int: ...
+    def wait_for_publish(self, timeout: float | None = ...) -> None: ...
+    def is_published(self) -> bool: ...
+
+class MQTTMessage:
+    timestamp: int
+    state: int
+    dup: bool
+    mid: int
+    payload: bytes | bytearray
+    qos: int
+    retain: bool
+    info: MQTTMessageInfo
+    properties: Properties | None
+    def __init__(self, mid: int = ..., topic: bytes | bytearray = ...) -> None: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    @property
+    def topic(self) -> str: ...
+    @topic.setter
+    def topic(self, value: bytes | bytearray) -> None: ...
+
+class Client:
+    suppress_exceptions: bool
+    def __init__(
+        self,
+        client_id: str = ...,
+        clean_session: bool | None = ...,
+        userdata: Any | None = ...,
```

Comment:
Optional thought: As `userdata` is used quite a bit below, maybe introduce a `_UserData` type alias to `Any`, just to make this explicit?